### PR TITLE
[Frame] Fix regression with sidebar styling

### LINKED ISSUE #11694: Frame causes sidebar overlap on apps
### Summary

A [recent PR](https://github.com/Shopify/polaris/pull/11628) that changed margin to padding on `Frame` styling caused a regression in Admin. Before, `Frame` components with the `sidebar` prop properly made room for the universal sidebar on app pages. Now, the `Frame` does not make room for the universal sidebar on app pages, even though the `sidebar` prop is set `true`.

### Expected behavior

Frame with sidebar should make room for the universal sidebar. See example below with padding changed to margin (undoing the change in the offending PR):

https://screenshot.click/06-40-xofhg-ionqi.png

### Actual behavior

Frame with sidebar does not make room for the universal sidebar:

https://screenshot.click/06-45-1tp2k-khcid.png

### Steps to reproduce

1. Open Admin 
2. Open universal sidebar by either:
  - Opening the Persistent Guide by clicking on a CTA in a Home Guide
  - Opening Sidekick
3. Go to an app page (e.g. Point of Sale, Email, etc.)
4. Verify that the `Frame` padding used to offset for the sidebar is not respected

### Are you using React components?

None

### Polaris version number

_No response_

### Browser

_No response_

### Device

_No response_

### PR DESCRIPTION
**Pull request by mmapplebeck**
A previous [PR](https://github.com/Shopify/polaris/pull/11628) attempting to fix an unwanted border next to the sidebar unintentionally caused a regression in Admin: app pages in Admin no longer maintain space on the right for the universal sidebar.

The fix: revert the change from the previous PR which changed margin to padding. Use margin, but instead of rendering `border-inline-end` on `.Content`, which causes the unwanted border, render it on `.Main` instead.

### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/11694

### WHAT is this pull request doing?

Before:

In Storybook, I added some styling to mimic how Admin apps are positioned absolutely. You can see how the sidebar overlaps the page content. This is what we want to fix.

https://screenshot.click/06-40-l8v5g-t7saz.png

After:

In Storybook, I added some styling to mimic how Admin apps are positioned absolutely. You can see how the sidebar spacing is respected and the sidebar does not overlap the content. The border is also correctly on the right side of the sidebar. `Frame` without sidebar also still correctly renders its border.

https://screenshot.click/06-45-4uju3-sn835.png

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Frame/Frame.module.scss
Comment (taha) [inline polaris-react/src/components/Frame/Frame.module.scss]: Would it be simpler to only have the border on `Main` instead of `Content`?

Comment (mmapplebeck) [reply polaris-react/src/components/Frame/Frame.module.scss]: Seems to work fine in Storybook testing. Will change it to be like this pending any feedback from Polaris reviewers that we need it on `.Content`

**Review by taha — APPROVED** [review]
LGTM to me! Pending review from the Polaris team.

**Comment by kyledurand** [comment]
Can you create a spin instance with the snapshot once it gets posted in a few mins?

File: .changeset/tame-hotels-repeat.md
Comment (kyledurand) [inline .changeset/tame-hotels-repeat.md]:
```suggestion
[Frame] Fixed regression that caused sidebar to render over Admin app content
```

**Review by kyledurand — APPROVED** [review]
This looks good and I can't find any issues in [Spin](https://admin.web.sidebar-test.kyle-durand.us.spin.dev/store/shop1)

**Comment by sophschneider** [comment]
hey @mmapplebeck! Just saw this introduced a line in the Admin, was this intentional? 

Screenshot 2024-04-17 at 4 13 50 PM
Screenshot 2024-04-17 at 4 13 21 PM

**Comment by mmapplebeck** [comment]
> hey @mmapplebeck! Just saw this introduced a line in the Admin, was this intentional?
> 
> Screenshot 2024-04-17 at 4 13 50 PM Screenshot 2024-04-17 at 4 13 21 PM

Not intentional. Will look into a fix shortly.

**Comment by mmapplebeck** [comment]
@sophschneider do you know if we even need this `border-inline-end: var(--p-border-width-025) solid var(--p-color-border);` style? It's been shifted around from `Content` to `Main`... but does this style serve any purpose that you know of? Can we just get rid of it?

**Comment by sophschneider** [comment]
@mmapplebeck I don't think so thats why I was asking since it seems to be added in this PR 😅 but now I see its just moved around here. Was it visible before this change? I feel like we could just remove it.

cc: @kyledurand do you have any idea if the border is doing anything?

**Comment by kyledurand** [comment]
I don't have any issue with removing it but can we leave it there until the new design ships? I'm not seeing it with the flag turned on in spin or storybook. If it's an issue maybe we can pair on it, or just assign it to me

**Comment by mmapplebeck** [comment]
@sophschneider @kyledurand  Let me know if/when I can help out with this. Otherwise, I'll leave it up to you all. Thanks! 🙏

**Pull request merged**
